src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs

```diff
@@ -210,6 +211,38 @@ public bool ExecuteReplicationOnce(OutgoingReplicationStatsScope stats, ref Date
                                 }
                                 lastTransactionMarker = item.TransactionMarker;
 
+                                if (_parent._inLegacyMode)
+                                {
```

Comment:
Wrap all of this in a separate method. In 99% of the cases, it won't be invoked, and we want to move it out from hot code